src/main/java/jchess/gamelogic/BoardActivity.java

```diff
@@ -116,6 +116,18 @@ public String toString() {
 						+ ", to:" 
 						+ m_oMoveCandidate.getCandidatePosition().getName();
 					break;
+				case "MOVE[KING_CASTLING]":
+					m_stToString 
+						= m_oPlayer.getName() 
+						+ " - from:" 
+						+ m_oMoveCandidate.getSourcePosition().getName() 
+						+ ", to:" 
+						+ m_oMoveCandidate.getCandidatePosition().getName()
+						+ " | - from:" 
+						+ m_oMoveCandidate.getSecondaryMoves().get(0).getSourcePosition().getName() 
```

Comment:
I think we should create a method in MoveCandidate class that should return SecondaryMove on behalf of the index provided, this way you can add try catch in it so that the system should not crash.